### PR TITLE
fix some issues of Array2

### PR DESCRIPTION
**Pull request by qindazhu**

### CODE REVIEW

File: k2/csrc/array.h [hunk 1]
```diff
@@ -112,21 +105,20 @@ struct Array2 {
   }
 
   IndexT size1;
+  IndexT size2;     // the number of elements in the array,  equal to
```
Comment (csukuangfj) [inline k2/csrc/array.h:108]: The initialization order of data members in the initialization list of the constructors should match the definition order.

**Comment by csukuangfj** [comment]
+2

File: k2/csrc/array.h (hunk 1 above)
Comment (qindazhu) [reply k2/csrc/array.h:108]: Actually I prefer to keep the original order (size1, indexes, size2, data) here as it's more readable (indexes is related to size1, and data is related to data).

Comment (csukuangfj) [reply k2/csrc/array.h:108]: Different people have different opinions about what is more readable.

The current approach is more memory-efficient.

Comment (qindazhu) [reply k2/csrc/array.h:108]: Will merge and leave it with `unresolved`. We can update it later if other guys (@danpovey) think we should change the order in the initialization list

**Comment by danpovey** [comment]
I agree with haowen, more memory efficient.

**Comment by csukuangfj** [comment]
> I agree with haowen, more memory efficient.

I am confused.

**Comment by csukuangfj** [comment]
> The initialization order of data members in the initialization list of the constructors should match the definition order.

It is a best practice to let the initialization order match the definition order,  though the current implementation is fine.

**Comment by danpovey** [comment]
Oh sorry.. Make it match to avoid warnings is what i would do.

**Comment by qindazhu** [comment]
There's no any warning in current code. 
Anyway, I'll change the initialization order in another PR as I don't want to take more time to argue about what is more readable here.

**Pull request merged**